packages/blade/src/components/StepGroup/_decisions/decisions.md

```diff
@@ -0,0 +1,219 @@
+    stepProgress="full"
+    trailing={<Badge>Label</Badge>}
+  />
+  {/* Nested StepGroup */}
+  <StepGroup>
+    <StepItem
+      leading={<StepItemIcon icon={CheckIcon} color="primary" />}
+      title="Needs Response"
+      timestamp="Respond latest by Tue, 23rd Oct'24 | 12:00pm"
+      description="Description"
+      stepProgress="start"
+    />
+    <StepItem
+      leading={<StepItemIcon icon={CheckIcon} color="neutral" />}
+      title="Needs Response"
+      timestamp="Respond latest by Tue, 23rd Oct'24 | 12:00pm"
+      description="Description"
+    />
+  </StepGroup>
+  <StepItem
+    leading={<StepItemIcon icon={CheckIcon} color="neutral" />}
+    title="Decision from Bank"
+    trailing={<Badge color="neutral">Pending</Badge>}
+  />
+</StepGroup>
+```
+
+### Horizontal Orientation
+
+<img width="400px" src="./image.png" />
+
+```jsx
+<StepGroup orientation="horizontal">{/* StepItem components */}</StepGroup>
+```
+
+## Accessibility
+
+- Make all steps visitable with `TAB` in interactive StepGroup
+  - Render StepItem as `a` tag when `href` is used
+  - Render StepItem as `button` tag when `onClick` is used without `href`
+- StepGroup does not need any separate role to be defined or special tag to be used.
+
+## References
+
+- [ProgressSteps - Baseweb](https://baseweb.design/components/progress-steps/)
+- [React Stepper - MUI](https://mui.com/material-ui/react-stepper/)
+- [Stepper - ZenDesk](https://garden.zendesk.com/components/stepper)
+
+All follow a similar compound API.
+
+## Open Questions
+
+- ### Should we add `isInteractive` prop?
+
+  In the proposed API, I have proposed that we can turn item into interactive or static based on whether it has `onClick` or `href` or nothing. Is it intuitive enough? or should we add more explict prop called `isInteractive` like we have in design
+
+- ### Alternative to `leading`
+
+  Currently I have proposed `leading` prop where we can add Icon or indicator. Although in horizontal orientation, its not exactly "leading". It comes on top. Its also not very equivalent to leading we have in other components.
+
+  Alternatives
+
+  - `marker={<StepItemIndicator color="positive" />}`
+
+- ### onClick and isSelected on StepItem vs selected and onItemClick on StepGroup
+
+  This is what I proposed primarily in this doc. To handle selections and clicks on Item alone
+
+  ```jsx
+  <StepItem
+    isSelected={selectedIndex === 0}
+    onClick={() => setSelectedIndex(0)}
+  >
+  ```
+
+  Another approach to this would be -
+
+  ```jsx
+  <StepGroup selected={selectedIndex} onItemClick={({ clickedItemIndex }) => {}}>
+    <StepItem />
+    <StepItem />
+  </StepGroup>
+  ```
+
+  An approach that MUI takes is mix of above 2 approaches -
+
+  ```jsx
+  <StepGroup selected={selectedIndex}>
+    <StepItem onClick={() => setSelecteIndex(0)} />
+    <StepItem onClick={() => setSelecteIndex(1)} />
+  </StepGroup>
+  ```
+
+  Q. Should we go with 1st, 2nd, or 3rd approach?
```

Comment:
Yup I am also leaning towards 1st one because of flexibility. I don't think so multiple selections is a usecase but that additional condition seems fair.

Also biggest reason is if we go with 3rd or 2nd, we have to define how we want to handle nested + interactive. So might have to define some strange index like `selected="1-2"`

Which is not the concern in 1st - https://github.com/razorpay/blade/pull/2124#discussion_r1562425878